### PR TITLE
Upgrade and fix nft client

### PR DESCRIPTION
**Pull request by sddioulde**
This package is broken. This PR fixes it.

Goal is that all future Audius collectibles logic uses this package as source of truth. e.g. https://github.com/AudiusProject/audius-protocol/pull/7991 consumes the version published by this PR's changes.

- Use Opensea v2
- Introduce Helius DAS API for solana nfts
- Introduce providers for collectibles from different chains, which depend on respective api clients
- Move and rename files
- Upgrade and publish to npm. Ready for consumption by clients.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -26,54 +26,68 @@ npm install @audius/fetch-nft
...
-type SolanaClientProps = {
-  rpcEndpoint?: string
-}
+  apiEndpoint?: string;
+  apiKey?: string;
+  assetLimit?: number;
+  eventLimit?: number;
+};
+
+type HeliusClientCtorProps = {
```
Comment (dylanjeffers) [inline README.md:59]: HeliusConfig?

File: src/eth/opensea.ts [hunk 2]
```diff
@@ -0,0 +1,114 @@
+import { OpenSeaCollection, OpenSeaEvent, OpenSeaNft } from './types'
+
+const OPENSEA_API_URL = 'https://api.opensea.io'
+const OPENSEA_NUM_ASSETS_LIMIT = 200
+const OPENSEA_NUM_EVENTS_LIMIT = 50
+
+export type OpenSeaClientCtorProps = {
```
Comment (dylanjeffers) [inline src/eth/opensea.ts:7]: nit but maybe "OpenSeaConfig"

File: src/eth/provider.ts [hunk 3]
```diff
@@ -0,0 +1,327 @@
...
+          ({ result, wallet }) =>
+            (result as PromiseFulfilledResult<OpenSeaNft[]>).value?.map(
+              (nft) => ({ ...nft, wallet })
+            ) || []
+        )
+        .flat()
+    )
+  }
+
+  // private async getNftTransferEventsForMultipleWallets(
```
Comment (dylanjeffers) [inline src/eth/provider.ts:47]: comment

File: src/eth/provider.ts [hunk 4]
```diff
@@ -0,0 +1,327 @@
...
+
+  async getCollectionMetadatasForCollectibles(
+    collectibles: Collectible[]
+  ): Promise<Collectible[]> {
+    // Build a set of collections to fetch metadata for
+    // and fetch them all at once, making sure to not fetch
+    // the same collection metadata multiple times.
+    const collectionSet = new Set<string>()
+    const idToCollectionMap = collectibles.reduce((acc, curr) => {
+      // Believe it or not, sometimes, rarely, the type of collection is an object
```
Comment (dylanjeffers) [inline src/eth/provider.ts:110]: lol

File: src/sol/provider.ts [hunk 5]
```diff
@@ -0,0 +1,178 @@
...
+          [
+            Buffer.from('metadata'),
+            this.metadataProgramIdPublicKey.toBytes(),
+            new PublicKey(mintAddress).toBytes()
+          ],
+          this.metadataProgramIdPublicKey
+        )
+      )[0]
+
+      const { Metadata } = await import(
```
Comment (dylanjeffers) [inline src/sol/provider.ts:159]: nice

**Review by dylanjeffers — APPROVED** [review]
amazing lift!

File: src/eth/provider.ts (hunk 3 above)
Comment (sddioulde) [reply src/eth/provider.ts:47]: on purpose lol. at least for now. this calls allows us to get a few more nfts, but the perf implication is heavy. so we decided to go with this for now. maybe we also simply remove the comment? no strong opinions there

Comment (sddioulde) [reply src/eth/provider.ts:47]: you know what. let's remove the comments. we have git history for reference, in case we want to add it back

**Pull request merged**